cmake/DevModeWarnings.cmake

```diff
@@ -107,6 +107,13 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     CCACHE_COMPILER_WARNINGS "-Wno-undefined-func-template")
   add_compile_flag_if_supported(
     CCACHE_COMPILER_WARNINGS "-Wno-return-std-move-in-c++11")
+
+  # -Wthread-safety-analysis (on FreeBSD when dev mode is enabled) gives false
+  # positive results currently. See PR #730.
```

Comment:
I'm not sure if it makes sense to apply this just to compensate for the current state in FreeBSD. As you say, maybe just don't build with `-Werror` and/or `-Weverything`?

Reply:
Disabling -Weverything seems like a big sacrifice. Removing -Werror sounds better to me.

Reply:
I don't want to remove `-Werror` from the developer builds simply because I like the developer to notice good warnings right away.

From https://clang.llvm.org/docs/UsersManual.html#diagnostics-enable-everything: _"Since -Weverything enables every diagnostic, we generally don’t recommend using it. -Wall -Wextra are a better choice for most projects. Using -Weverything means that updating your compiler is more difficult because you’re exposed to experimental diagnostics which might be of lower quality than the default ones."_ [Others agree](https://quuxplusone.github.io/blog/2018/12/06/dont-use-weverything/).